### PR TITLE
N21-539 migration error page

### PR DESCRIPTION
**Pull request by MarvinOehlerkingCap**
# Short Description
- adds a page that is displayed in case the user login migration fails

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/N21-539

## Screenshots of UI changes
image

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/locales/de.json [hunk 1]
```diff
@@ -889,9 +889,12 @@
...
 	"pages.userMigration.description.fromSource": "Deine Schule wechselt gerade das Anmeldesystem von {sourceSystem} zu {targetSystem} und Du kannst Dein Anmeldeverfahren von {sourceSystem} zu {targetSystem} umziehen lassen.<br><br>Du kannst den Umzug jetzt starten. Zu Beginn des Umzugs meldest Du Dich ein letztes Mal mit {sourceSystem} an. Danach kannst Du Dich nur noch über {targetSystem} anmelden.",
 	"pages.userMigration.description.fromSourceMandatory": "Deine Schule wechselt gerade das Anmeldesystem von {sourceSystem} zu {targetSystem} und damit zieht auch Dein Anmeldeverfahren endgültig um.<br><br>Du musst den Umzug jetzt starten. Zu Beginn des Umzugs meldest Du Dich ein letztes Mal mit {sourceSystem} an. Danach kannst Du Dich nur noch über {targetSystem} anmelden.",
+	"pages.userMigration.error.title": "Umzug des Anmeldesystems ist fehlgeschlagen",
+	"pages.userMigration.error.description": "Der Umzug des Anmeldesystems von {sourceSystem} zu {targetSystem} ist leider fehlgeschlagen. Um die Niedersächsische Bildungscloud mit dem Anmeldesystem {targetSystem} nutzen zu können, bitte <a href=\"{supportLink}\">Kontakt zum Support</a> aufnehmen.",
```
Comment (arnegns) [inline src/locales/de.json:897]: There is a placeholder for the instance name.. maybe you could use this for "Niedersächsische Bildungscloud"

File: src/pages/user-migration/UserLoginMigrationError.page.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		</h1>
+		<div>
+			<p
+				class="pa-4"
+				data-testId="text-description"
+				v-html="
+					$t('pages.userMigration.error.description', {
+						sourceSystem: getSystemName(sourceSystem),
+						targetSystem: getSystemName(targetSystem),
+						supportLink:
```
Comment (arnegns) [inline src/pages/user-migration/UserLoginMigrationError.page.vue:18]: We could use the env: ACCESSIBILITY_REPORT_EMAIL .. we only need the to pass this in dof app deploy to nuxt. I think this should be nicer

File: src/router/routes.ts [hunk 3]
```diff
@@ -216,4 +222,32 @@ export const routes: Array<RouteConfig> = [
 			layout: Layouts.LOGGED_OUT,
 		},
 	},
+	{
+		path: "/migration/error",
+		component: () =>
+			import("@/pages/user-migration/UserLoginMigrationError.page.vue"),
+		name: "user-login-migration-error",
+		beforeEnter: (to: Route, from: Route, next: NavigationGuardNext) => {
```
Comment (arnegns) [inline src/router/routes.ts:230]: Like the other PR. This should be moved to a composable or in the component itself if we cant reuse this logic

**Review by arnegns — CHANGES_REQUESTED** [review]

File: src/pages/user-login-migration/UserLoginMigrationError.page.vue [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<template>
+	<div v-show="hasData" class="text-center mx-auto container-max-width">
+		<img
+			src="@/assets/img/migration/migration_error.svg"
+			alt="migration error logo"
```
Comment (arnegns) [inline src/pages/user-login-migration/UserLoginMigrationError.page.vue:5]: Maybe we can translate alt's

File: src/router/guards/query-parameter.guard.ts [hunk 5]
```diff
@@ -8,7 +8,7 @@ const { createApplicationError } = useApplicationError();
 
 type ParameterValidationFunction = (value: unknown, to: Route) => boolean;
 
-export function createQueryParameterGuard(
+export function createQueryParameterValidationGuard(
```
Comment (arnegns) [inline src/router/guards/query-parameter.guard.ts:11]: Can you rename also the file to validate-query-parameter.guard?

File: src/store/env-config.ts [hunk 6]
```diff
@@ -111,25 +111,35 @@ export default class EnvConfigModule extends VuexModule {
 	get getTeacherStudentVisibilityIsConfigurable() {
 		return this.env.TEACHER_STUDENT_VISIBILITY__IS_CONFIGURABLE;
 	}
+
```
Comment (arnegns) [inline src/store/env-config.ts:114]: Nice :)

**Review by arnegns — DISMISSED** [review]

File: src/pages/user-login-migration/UserLoginMigrationError.page.vue [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+		const getSystemName = (id: string): string => {
+			return (
+				systemsModule?.getSystems.find(
+					(system: System): boolean => system.id === id
+				)?.name ?? ""
+			);
+		};
+
+		const hasData: Ref<boolean> = ref(false);
```
Comment (OliverHappe) [inline src/pages/user-login-migration/UserLoginMigrationError.page.vue:70]: small suggestions: call this isLoading and init with true. isLoading is used all the time across the app to convey the state where we wait for data to load.

Comment (MarvinOehlerkingCap) [reply src/pages/user-login-migration/UserLoginMigrationError.page.vue:70]: Do I understand it right that I should rename hasData to isLoading?

**Review by OliverHappe — APPROVED** [review]

**Pull request merged**